netbout-web/src/main/java/com/netbout/rest/TkStart.java

```diff
@@ -67,13 +81,35 @@ public TkStart(final Base bse) {
 
     @Override
     public Response act(final Request req) throws IOException {
-        final Inbox inbox = new RqAlias(this.base, req).alias().inbox();
+        final Href href = new RqHref.Base(req).href();
+        final Iterator<String> token = href.param("token").iterator();
+        String key = "";
+        final Inbox inbox;
+        while (token.hasNext()) {
+            key = token.next();
+        }
+        if (key.isEmpty()) {
+            inbox = new RqAlias(this.base, req).alias().inbox();
+        } else if (this.tokens.containsKey(key)) {
+            final Pair<Date, Inbox> cached = this.tokens.get(key);
+            final long diff = (new Date()).getTime()
+                - (cached.getLeft()).getTime();
+            if ((TimeUnit.MILLISECONDS.toDays(diff)) <= 2) {
+                inbox = cached.getRight();
+            } else {
+                this.tokens.remove(key);
+                inbox = new RqAlias(this.base, req).alias().inbox();
+            }
+        } else {
+            inbox = new RqAlias(this.base, req).alias().inbox();
+            this.tokens.put(key,
+                    Pair.of(new Date(), inbox));
```

Comment:
@prahladyeri Let's move this param to the previous line, it shouldn't exceed length limit anymore.